tensorflow/core/kernels/group_lookup_forward_ops.cc

```diff
@@ -306,6 +370,36 @@ class GroupEmbeddingVariableLookupCpuOp : public OpKernel {
             [this, &gather_embedding, batch_nums, unique_idx, unique,
              unique_embedding_data, sp_weights](int64 start, int64 end) {
               for (int64 i = start; i < end; ++i) {
+#if defined(__GNUC__) && (__GNUC__ > 6) && (__AVX512F__)
+                int tmp_length = (dimension_ + 15) / 16;
+                __m512 tmp_embedding[tmp_length];
+                for(int i = 0; i < tmp_length; ++i){
+                  tmp_embedding[i] = _mm512_set1_ps(0.0f);
+                }
+                int batch_offset = i == 0 ? 0 : batch_nums[i - 1];
+                int batch_num = batch_nums[i] - batch_offset;
+                for (int j = 0; j < batch_num; ++j) {
+                  int unique_indice = unique_idx[batch_offset + j];
+                  float* u_embedding =
+                      unique_embedding_data + unique_indice * dimension_;
+                  __m512 _weights = _mm512_set1_ps(*(sp_weights + batch_offset + j));
+                  for (int d = 0; d < dimension_; d += 16) {
+                    int index = d / 16;
+                    int remain = dimension_ - d;
+                    __mmask16 mask = (remain >= 16 ? 0xffff : (1 << remain) - 1);
+                    __m512 _item = _mm512_maskz_loadu_ps(mask, u_embedding + d);
+                    tmp_embedding[index] = _mm512_mask3_fmadd_ps(_item, _weights, tmp_embedding[index], mask);
+                  }
+                }
+                __m512 _bs = _mm512_set1_ps(sqrtf(batch_num));
+                for (int d = 0; d < dimension_; d += 16) {
+                  int index = d / 16;
+                  int remain = dimension_ - d;
+                  __mmask16 mask = (remain >= 16 ? 0xffff : (1 << remain) - 1);
+                  tmp_embedding[index] = _mm512_div_ps(tmp_embedding[index], _bs);
```

Comment:
this operation can be merged into the fma step, the same as doing weights multiplication.